lms/validation/authentication/_bearer_token.py

```diff
@@ -36,55 +31,35 @@ class BearerTokenSchema(PyramidRequestSchema):
         'Bearer eyJ...YoM'
 
         >>> # Deserialize the request's authorization param into an LTI user.
-        >>> schema.lti_user()
-        LTIUser(user_id='...', application_instance_id='...', ...)
-
-    The above are convenience methods that wrap webargs and marshmallow. But
-    this class is also a marshmallow schema and can be used via the usual
-    webargs or marshmallow APIs. For example to serialize a dict and then
-    deserialize the same dict back from its serialized form using marshmallow::
-
-        >>> schema.dump(request.lti_user).data
-        {'authorization': 'Bearer eyJ...YoM'}
-
-        >>> schema.load({'authorization': 'Bearer eyJ...YoM'}).data
-        LTIUser(user_id='...', application_instance_id='...', ...)
-
-    Or to parse an models.LTIUser out of a Pyramid
-    request's authorization param using webargs::
-
-        >>> from webargs.pyramidparser import parser
-        >>> parser.parse(s, request)
+        >>> schema.lti_user(location='headers')
         LTIUser(user_id='...', application_instance_id='...', ...)
     """
 
-    user_id = marshmallow.fields.Str(required=True)
-    application_instance_id = marshmallow.fields.Int(required=True)
-    roles = marshmallow.fields.Str(required=True)
-    tool_consumer_instance_guid = marshmallow.fields.Str(required=True)
-    display_name = marshmallow.fields.Str(required=True)
-    email = marshmallow.fields.Str()
+    authorization = marshmallow.fields.Str(required=True)
```

Comment:
Changed this schema to behave like:

https://github.com/hypothesis/lms/blob/2fff474202d162dc3787354bea0425bb7b0f3cba/lms/validation/authentication/_oauth.py

I does have less features, it does not validate the actual fields of LTIUser but this is data we send to ourselves rather than user input. 

IMO Missing/wrong data here should raise and exception in sentry (like it will in this version) instead of being masked as 403/401 and lost in the monitoring.

Reply:
As a summary for myself, the change seems to be:

Before we used lots of marshmallow magic to move directly from the bearer token to the decomposed elements packed inside the JWT as attributes on schema. These attributes aren't actually used outside of this class (despite being public) and are then used to reasonably trivially create an `LTIUser` object.

Now we go part of the way by storing the token as a single attribute `authorization`, and then perform the JWT decomposition on demand when accessing the `LTIUser` object.

As a change in behavior we now no longer validate the contents of the JWT to ensure they form a valid user. We hope that any problems caused by missing elements will show up elsewhere in the code.

Reply:
```suggestion
    _jwt_token = marshmallow.fields.Str(required=True, data_key="authorization")
```

Is this a private attribute? I don't think we actually do / should access it from outside this class? The fact we are a schema is actually a private matter I think?